src/main/java/io/openliberty/tools/common/plugins/config/ServerConfigDocument.java

```diff
@@ -559,17 +616,17 @@ private static void parseIncludeVariables(Document doc) throws XPathExpressionEx
 
                 if (includeFileName == null || includeFileName.trim().isEmpty()) {
                     log.warn("Unable to resolve include file location "+nodeValue+". Skipping the included file during application location processing.");
-                    return;
+                    continue;
                 }
-                    
-                Document docIncl = getIncludeDoc(includeFileName);
-
-                if (docIncl != null) {
-                    parseVariablesForBothValues(docIncl);
-                    // handle nested include elements
-                    parseIncludeVariables(docIncl);
-                } else {
-                    log.warn("Unable to parse include file "+includeFileName+". Skipping the included file during application location processing.");
+
+                ArrayList<Document> inclDocs = getIncludeDoc(includeFileName);
+
+                for (Document inclDoc : inclDocs) {
+                    if (inclDoc != null) {
```

Comment:
inclDoc cannot be null. Should probably check for an empty list and log a warning.

Reply:
the `getIncludeDoc()` method logs a warning before returning the empty list.
Removing null check here